dev/com.ibm.ws.kernel.service/src/com/ibm/ws/kernel/service/location/internal/WsLocationAdminImpl.java

```diff
@@ -335,7 +339,26 @@ protected WsLocationAdminImpl(Map<String, Object> config) {
 
         addResourcePath(bootstrapLib.getNormalizedPath());
 
-        SymbolRegistry.getRegistry().addStringSymbol(WsLocationConstants.LOC_VARIABLE_SOURCE_DIRS, (String) config.get(WsLocationConstants.LOC_VARIABLE_SOURCE_DIRS));
+        final String variableSourceDirs = (String) config.get(WsLocationConstants.LOC_VARIABLE_SOURCE_DIRS);
+        SymbolRegistry.getRegistry().addStringSymbol(WsLocationConstants.LOC_VARIABLE_SOURCE_DIRS, variableSourceDirs);
+
+        if (bundleContext != null) {
+            bundleContext.registerService(CheckpointHookFactory.class, (p) -> new CheckpointHook() {
+                @Override
+                public void restore() {
+                    String newVariableSourceDirs = getEnv(BootstrapConstants.ENV_VARIABLE_SOURCE_DIRS);
+                    if (newVariableSourceDirs != null && !newVariableSourceDirs.equals(variableSourceDirs)) {
```

Comment:
What about when `newVariableSourceDirs` is null.  I think there is a default location we would need to restore to.

Reply:
I fixed the whole method. The way I was treating the environment variable earlier was wrong. There are multiple paths allowed in that variable.